WordPress/src/test/java/org/wordpress/android/ui/jetpack/scan/details/ThreatDetailsListItemsBuilderTest.kt

```diff
@@ -93,182 +92,319 @@ class ThreatDetailsListItemsBuilderTest : BaseUnitTest() {
+        assertThat(threatItems.map { it::class.java }).containsSequence(
+            HeaderState::class.java,
+            DescriptionState::class.java,
+            ThreatFileNameState::class.java,
+            DescriptionState::class.java
         )
     }
 
     @Test
-    fun `builds technical details list items correctly for CoreFileModificationThreatModel`() {
-        // Arrange
-        val expectedTechnicalDetailsHeaderItem = technicalDetailsHeaderItem
-        val expectedFileNameDescriptionItem = fileNameDescriptionItem
-        val expectedFileNameItem = ThreatFileNameState(
-            UiStringText(ThreatTestData.coreFileModificationThreatModel.fileName)
-        )
-        val expectedDiffItem = DescriptionState(UiStringText(ThreatTestData.coreFileModificationThreatModel.diff))
+    fun `given core file modif threat, when items are built, then technical details header exists`() {
+        // Act
+        val threatItems = buildThreatDetailsListItems(ThreatTestData.coreFileModificationThreatModel)
 
+        // Assert
+        assertThat(threatItems).contains(technicalDetailsHeaderItem)
+    }
+
+    @Test
+    fun `given core file modif threat with file name, when items are built, then file name description exists`() {
         // Act
         val threatItems = buildThreatDetailsListItems(ThreatTestData.coreFileModificationThreatModel)
 
         // Assert
-        Assertions.assertThat(threatItems).containsSubsequence(
-            expectedTechnicalDetailsHeaderItem,
-            expectedFileNameDescriptionItem,
-            expectedFileNameItem,
-            expectedDiffItem
+        assertThat(threatItems).contains(fileNameDescriptionItem)
+    }
+
+    @Test
+    fun `given core file modif threat with file name, when items are built, then file name exists`() {
+        // Act
+        val threatItems = buildThreatDetailsListItems(ThreatTestData.coreFileModificationThreatModel)
+
+        // Assert
+        assertThat(threatItems).contains(fileNameDescriptionItem)
+    }
+
+    @Test
+    fun `given core file modif threat, when items are built, then diff item exists`() {
+        // Act
+        val threatItems = buildThreatDetailsListItems(ThreatTestData.coreFileModificationThreatModel)
+
+        // Assert
+        assertThat(threatItems).contains(
+            DescriptionState(UiStringText(ThreatTestData.coreFileModificationThreatModel.diff))
         )
     }
 
+    /* FIXABLE THREAT */
+
     @Test
-    fun `builds fix details items correctly for a fixable threat`() {
+    fun `given fixable threat not in fixed status, when items are built, then fix header exists`() {
         // Arrange
         val fixableThreat = GenericThreatModel(
             ThreatTestData.baseThreatModel.copy(
-                fixable = Fixable(file = null, fixer = Fixable.FixType.EDIT, target = null),
-                status = ThreatStatus.CURRENT
+                fixable = Fixable(file = null, fixer = Fixable.FixType.EDIT, target = null)
```

Comment:
Let me know if we should add separate tests for each `FixType` for this and the following test for fix header and description items.

Reply:
It's difficult for me to evaluate that. It really depends on how much value do you think it'd add and if it takes 5 minutes or 5 hours. :) Use your best judgment.